### DISCUSSION TITLE
[Hacktoberfest Idea] PyPI RSS Feed analyzer to automate plugin testing matrix

### DISCUSSION THREAD

**Discussion by Superskyyy** [apache/skywalking#9629] [Hackathon Ideas (Newcomers look here)]
The current SkyWalking Python agent uses a matrix to define its plugin tests (which python version tests which framework version). However, it's beyond maintainers' reasonable effort to constantly update and monitor this matrix. As a result, we [don't update the supported plugin docs in time](https://skywalking.apache.org/docs/skywalking-python/next/en/setup/plugins/), and new users may be confused about whether it will work for their applications. It forces users to do trial-and-error experiments, which is a big no-no in my opinion.

Usually, these frameworks evolve rather quickly (especially in Python) and may introduce major changes that could break the plugins. We know all the supported plugins are targeted for PyPI listed packages, and they all have an [RSS feed ](https://pypi.org/rss/project/poetry/releases.xml) to indicate the latest version number, supported Python versions etc. Let's build a scraper on GitHub actions to check for this and update the testing matrix periodically (maybe on new major versions e.g. 1.8.1 -> 2.0.0). This way, we can realize breaking changes and automatically update the range of supported versions in documentation site.

Something like the below flow will be sufficient. 
![image](https://user-images.githubusercontent.com/26076517/190921373-bc19facf-d972-462a-abed-7b38b0c06bcb.png)

Involved technology (Skills):

1. Some familiarity with CI automation (GitHub actions)
2. Python requests or similar library and basic structured text analysis skills
3. Git

If successful, this idea could be expanded to multiple agents.

**Comment by Superskyyy** [comment]
If anyone would like to take this, please provide a brief proposal (Google Docs) and describe how you plan to implement it.

**Comment by Superskyyy** [comment]
The sraping part is being worked on.

**Reply by kezhenxu94** [reply]
Who is working on this? And can you paste the Google doc lint here?
